test/e2e/serial/tests/non_regression.go

```diff
@@ -191,5 +195,123 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
 				return ok
 			}, time.Minute, time.Second*5).Should(BeTrue(), "resources not restored on %q", targetedNodeName)
 		})
+
+		It("[test_id:48964][tier2] should be able to schedule a guaranteed deployment pod in selective way", func() {
+			nrtInitialList := nrtv1alpha1.NodeResourceTopologyList{}
+
+			err := fxt.Client.List(context.TODO(), &nrtInitialList)
+			Expect(err).ToNot(HaveOccurred())
+
+			workers, err := e2enodes.GetWorkerNodes(fxt.Client)
+			Expect(err).ToNot(HaveOccurred())
+
+			// TODO choose randomly
+			targetNodeName := workers[0].Name
+
+			By(fmt.Sprintf("explicitly mentioning which we want pod to land on node %q", targetNodeName))
+
+			// By("setting a fake schedule name under the pod to make sure pod not scheduled by any scheduler")
+			nonExistingSchedulerName := "foo"
+
+			var replicas int32 = 1
+			podLabels := map[string]string{
+				"test": "test-dp",
+			}
+			nodeSelector := map[string]string{}
+			// the pod is asking for 2 CPUS and 100Mi
+			requiredRes := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("2"),
+				corev1.ResourceMemory: resource.MustParse("100Mi"),
+			}
+			podSpec := &corev1.PodSpec{
+				SchedulerName: nonExistingSchedulerName,
+				Containers: []corev1.Container{
+					{
+						Name:    "testdp-cnt",
+						Image:   objects.PauseImage,
+						Command: []string{objects.PauseCommand},
+						Resources: corev1.ResourceRequirements{
+							Limits:   requiredRes,
+							Requests: requiredRes,
+						},
+					},
+				},
+				NodeName:      targetNodeName,
+				RestartPolicy: corev1.RestartPolicyAlways,
+			}
+
+			By(fmt.Sprintf("creating a deployment with a guaranteed pod with two containers requiring total %s", e2ereslist.ToString(e2ereslist.FromContainers(podSpec.Containers))))
+			dp := objects.NewTestDeploymentWithPodSpec(replicas, podLabels, nodeSelector, fxt.Namespace.Name, "testdp", *podSpec)
+
+			err = fxt.Client.Create(context.TODO(), dp)
+			Expect(err).ToNot(HaveOccurred())
+
+			updatedDp, err := e2ewait.ForDeploymentComplete(fxt.Client, dp, time.Second*10, time.Minute)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
if we select a non esistent scheduler, why the pods managed to land on a node? let's add a comment to clarify this

Reply:
Yeah, good point. I think I referred to another test where something similar was done. I will double check this and make sure this does not cause unexpected behavior. Would probably be better to use the default scheduler here instead.

Reply:
I updated this and am no longer providing a `schedulerName` explicitly so that the pod is scheduled by the default scheduler.